### ISSUE TITLE
Add event logging and log related behaviours

### ISSUE THREAD

**Issue by carlnewton** [carlnewton/habitat#20]

**Comment by carlnewton** [comment]
If a user has posted more than x times in y minutes, they should not be able to post again z minutes
If a user has commented more than x times in y minutes, they should not be able to comment again for z minutes
If a user has failed to login with the same account more than x times in y minutes, they should not be able to attempt again*
If an IP address has failed to login with any account more than x times in y minutes, they should not be able to attempt again*


*this needs to be thought out further

**Comment by ahmedfaarih** [comment]
Hey, just want to confirm before I start working on this — this is basically rate limiting for posts/comments and login attempts right? I can use Symfony's RateLimiter component for this. Let me know if I'm on the right track.

**Comment by carlnewton** [comment]
Yes, but I think it might be best to hold off from this for now. I think this might not work well when a proxy is used and the end user IP address is not forwarded to the application. Though if symfony rate limiting is done by sessions it might be okay.
